airflow/models/taskinstance.py

```diff
@@ -3476,8 +3476,13 @@ def _schedule_downstream_tasks(
                     run_id=ti.run_id,
                 ),
                 session=session,
-                nowait=True,
-            ).one()
+                skip_locked=True,
+            ).one_or_none()
+
+            if not dag_run:
+                cls.logger().info("Skip locked rows, rollback")
+                session.rollback()
```

Comment:
Why are we doing rollback here?

Reply:
just a [similar](https://github.com/apache/airflow/blob/61e90704c6794e8554d17a8a94a6e3f42f304767/airflow/models/taskinstance.py#L3531) pattern used in this method on error.

Reply:
The "schedule_downstream_tasks" is only executed in a new session, so rollback is safe here, just in case.